context/app/static/js/components/workspaces/NewWorkspaceDialog/NewWorkspaceDialog.tsx

```diff
@@ -0,0 +1,202 @@
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Paper from '@mui/material/Paper';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import Stack from '@mui/material/Stack';
+import Chip, { ChipProps } from '@mui/material/Chip';
+import Button from '@mui/material/Button';
+
+import Step from 'js/shared-styles/surfaces/Step';
+import ContactUsLink from 'js/shared-styles/Links/ContactUsLink';
+import SelectableChip from 'js/shared-styles/chips/SelectableChip';
+import { useSelectItems } from 'js/hooks/useSelectItems';
+import MultiAutocomplete from 'js/shared-styles/inputs/MultiAutocomplete';
+import WorkspaceField from 'js/components/workspaces/WorkspaceField';
+import TemplateGrid from '../TemplateGrid';
+import { useWorkspaceTemplates, useWorkspaceTemplateTags } from './hooks';
+import { CreateWorkspaceFormTypes } from './useCreateWorkspaceForm';
+import { CreateTemplateNotebooksTypes } from '../types';
+
+interface TagTypes extends ChipProps {
+  option: string;
+}
+
+function TagComponent({ option, ...rest }: TagTypes) {
+  return <Chip label={option} {...rest} />;
+}
+
+type ReactHookFormProps = Pick<UseFormReturn<CreateWorkspaceFormTypes>, 'handleSubmit' | 'control'> & {
+  errors: FieldErrors<CreateWorkspaceFormTypes>;
+};
+
+interface NewWorkspaceDialogProps {
+  datasetUUIDs: Set<string>;
+  errorMessages: string[];
+  dialogIsOpen: boolean;
+  handleClose: () => void;
+  onSubmit: ({ workspaceName, templateKeys, uuids }: CreateTemplateNotebooksTypes) => void;
+}
+
+const recommendedTags = ['visualization', 'api'];
+
+function NewWorkspaceDialog({
+  datasetUUIDs,
+  errorMessages = [],
+  dialogIsOpen,
+  handleClose,
+  handleSubmit,
+  control,
+  errors,
+  onSubmit,
+  children,
+}: PropsWithChildren<NewWorkspaceDialogProps & ReactHookFormProps>) {
+  const { selectedItems: selectedRecommendedTags, toggleItem: toggleTag } = useSelectItems([]);
+  const [selectedTags, setSelectedTags] = useState<string[]>([]);
+
+  const { templates } = useWorkspaceTemplates([...selectedTags, ...selectedRecommendedTags]);
+
+  const { tags } = useWorkspaceTemplateTags();
+
+  const submit = useCallback(
+    ({ 'workspace-name': workspaceName, templates: templateKeys }: CreateWorkspaceFormTypes) => {
+      onSubmit({
+        workspaceName,
+        templateKeys,
+        uuids: [...datasetUUIDs],
+      });
+    },
+    [datasetUUIDs, onSubmit],
+  );
+
+  return (
+    <Dialog
+      open={dialogIsOpen}
+      onClose={handleClose}
+      scroll="paper"
+      aria-labelledby="scroll-dialog-title"
+      aria-describedby="scroll-dialog-description"
+      maxWidth="lg"
+    >
+      <Box mb={2}>
+        <DialogTitle id="scroll-dialog-title" variant="h3">
+          Launch New Workspace
+        </DialogTitle>
+        <Box sx={{ px: 3 }}>
+          <Paper sx={{ p: 2 }}>
+            <Typography sx={{ mb: 2 }}>
```

Comment:
I think if we use `<Stack gap={2} sx={{ p: 2 }} component={Paper} direction="col">` we could cut out the inner `sx={{ mb: 2 }}` here; I think we could also add `mx: 3` and remove the outer wrapping Box on line 89 as well, without affecting styles.